### PR TITLE
Feat: add multi language for services name

### PR DESCRIPTION
**Pull request by thecampagnards**
image

# Data in mongo : 
## functionalServices
```json
{ "_id" : ...., "name" : "....", "package" : "....", "position" : ...., "services" : [ "..." ], "translations" : [ { "translation" : "en translation .....", "languagecode" : "en" } ], "declarativeDeployement" : false }
```
## languages
```json
{ "_id" : ..., "languagecode" : "fr" }
```

### CODE REVIEW

File: client/src/components/projects/project/matrix/matrix.component.jsx [hunk 1]
```diff
@@ -52,6 +53,9 @@ class Matrix extends React.Component {
     const goalOption = options.find((elm) => elm.value === matrix.goal);
     const dueDate = matrix.dueDate ? moment(matrix.dueDate) : '';
     const expandComment = this.state && this.state.expandComment;
+    // Check if there is a language and translation and it's not the default language
+    // If there is any error the default translation will be choosed
```
Comment (SmalkiGroove) [inline client/src/components/projects/project/matrix/matrix.component.jsx]: ❗️ chosen

File: server/controllers/languages.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+// Languages is the controller type
+type Languages struct {
+}
+
+// GetAll languages from database
+func (u *Languages) GetAll(c echo.Context) error {
+	database := c.Get("database").(*mongo.DadMongo)
+	languages, err := database.Languages.FindAll()
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, types.NewErr("Error while retrieving all languages"))
```
Comment (SmalkiGroove) [inline server/controllers/languages.go]: Tu pourrais mettre un message plus clair mec, au moins dire que c'est un problème de connexion à la base mongo

File: client/src/components/projects/project/matrix/matrix.component.jsx (hunk 1 above)
Comment (thecampagnards) [reply client/src/components/projects/project/matrix/matrix.component.jsx]: ppppffff

File: server/controllers/languages.go (hunk 2 above)
Comment (thecampagnards) [reply server/controllers/languages.go]: c'est du ctrl c +v xD

File: server/controllers/languages.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	exists, err := database.Languages.Exists(language.LanguageCode)
+	if err != nil {
+		return c.JSON(http.StatusBadRequest, types.NewErr(fmt.Sprintf("Error while checking if the language already exists: %v", err)))
+	}
+	if exists {
+		return c.JSON(http.StatusConflict, types.NewErr(fmt.Sprintf("Received language already exists")))
+	}
+
+	// language will be created
+	language.ID = ""
```
[the remark this answers is not in the document]
Comment (thecampagnards) [reply server/controllers/languages.go]: ca separe la couche dao de la couche controller

File: server/controllers/languages.go [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+	languageSaved, err := database.Languages.Save(language)
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, types.NewErr(fmt.Sprintf("Failed to save language to database: %v", err)))
+	}
+
+	return c.JSON(http.StatusOK, languageSaved)
```
[the remark this answers is not in the document]
Comment (thecampagnards) [reply server/controllers/languages.go]: nop c'est loggé cote mongo et le mec aura un statut 200 avec son nouvel object

File: server/types/language.go [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+func (r *LanguageRepo) Save(language Language) (Language, error) {
+	if !r.isInitialized() {
+		return Language{}, ErrDatabaseNotInitialized
+	}
+
+	if language.ID.Hex() == "" {
+		language.ID = bson.NewObjectId()
+	}
+
+	_, err := r.col().UpsertId(language.ID, bson.M{"$set": language})
```
[the remark this answers is not in the document]
Comment (thecampagnards) [reply server/types/language.go]: https://godoc.org/gopkg.in/mgo.v2#Collection.UpsertId https://godoc.org/gopkg.in/mgo.v2#Collection.Upsert gosso modo avec Id ca simplif l'ecriture et upsert ca 
```Upsert finds a single document matching the provided selector document and modifies it according to the update document. If no document matching the selector is found, the update document is applied to the selector document and the result is inserted in the collection. If the session is in safe mode (see SetSafe) details of the executed operation are returned in info, or an error of type *LastError when some problem is detected.```

**Pull request merged**